### ISSUE TITLE
weird condition

### ISSUE THREAD

**Issue by yahyaKacem** [jamesallardice/jslint-error-explanations#27]
Hi, first thanks for this site.
Here's how this happened:
code with the error:

```
(function () {
  "use strict";
  var array = [1, 2, 3];
  if (!!array.indexOf(2)) {
    console.dir('in array');
  }
}());
```

After running jslint spits this error:

> 1 Weird condition.
>     if (!!array.indexOf(2)) { // Line 6, Pos 22

Changing the code to this fix it:

```
(function () {
  "use strict";
  var
    array = [1, 2, 3],
    test  = array.indexOf(2);
  if (!!test) {
    console.dir('in array');
  }
}());
```

Could you make a page about this I still don't know why using the first is a bad practice, or is't just a false error.
Thanks in advance.
